nixos/modules/system/boot/stage-1.nix

```diff
@@ -128,6 +128,13 @@ let
         copy_bin_and_libs ${pkgs.e2fsprogs}/sbin/resize2fs
       ''}
 
+      # Copy f2fs-tools' fsck and resize if needed
+      ${optionalString (any (fs: fs.autoResize) (filter (x: x.fsType == "f2fs") fileSystems)) ''
```

Comment:
I would say:

```
${optionalString (any (fs: fs.autoResize && fs.fsType = "f2f2") fileSystems)
```

Reply:
Oops. Didn't consider the obvious fact that && stands for logical AND operation. One sec, will push the updated version!

Reply:
If you are at it, please also move this code to: modules/tasks/filesystems/f2fs.nix

Reply:
@Mic92 moved to nixos/f2fs module.